### PR TITLE
Ghost cafe quality-of-life, plus making ghost role eligibility an element

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

1. Ghost cafe spawns can now ghostize in a way that lets them return to their body at any time.
2. The floor of the ghost cafe is now (mostly) indestructible.
3. There is now a cremator in the ghost cafe for all of your body disposal needs.
4. Ghost cafe spawns can now hear dead chat.
5. Ghost cafe spawns are now eligible for ghost role spawns, due to...
6. The fact that ghost role eligibility is now an element, keeping a list of all role-eligible mobs and all cooldowns before eligibility indexed by ckey.

## Why It's Good For The Game

Ghost cafe probably shouldn't be, like, an investment, or even a commitment. This makes it zero-commitment, essentially an alternate way to observe.

## Changelog
:cl:
add: Ghost cafe mobs can now ghostize
add: Ghost cafe now has a cremator
add: Ghost cafe mobs are now eligible for ghost roles
refactor: Ghost roles now use an element for eligibility purposes
/:cl:

### CODE REVIEW

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 1]
```diff
@@ -626,7 +626,11 @@
 		O.equip(new_spawn, FALSE, new_spawn.client)
 		SSjob.equip_loadout(null, new_spawn, FALSE)
 		SSquirks.AssignQuirks(new_spawn, new_spawn.client, TRUE, TRUE, null, FALSE, new_spawn)
+		new_spawn.AddElement(/datum/element/ghost_role_eligibility)
+		ADD_TRAIT(new_spawn, TRAIT_SIXTHSENSE, GHOSTROLE_TRAIT)
```
Comment (Ghommie) [inline code/game/objects/structures/ghost_role_spawners.dm:630]: Isn't the "ghost cafe" more or less the same location of the "ninja prison"?
Shouldn't a few chat lines to warn against blurting out OOC in IC or deadchat while inside the body be needed?

**Review by Ghommie — APPROVED** [review]
Finally, some good fucking code.

Also Travis failed because of the mispelled reagent IDs meme. Nothing to worry about.

**Comment by ghost** [comment]
congus

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 1 above)
Comment (Putnam3145) [reply code/game/objects/structures/ghost_role_spawners.dm:630]: Yeah, I'll do that with the merge conflict fix

**Comment by Tupinambis** [comment]
Excellent change. Goon features best features.

**Pull request merged**